File: mobile/src/main/java/net/maiatoday/levelbest/di/ApplicationModule.java
package net.maiatoday.levelbest.di;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import com.google.firebase.analytics.FirebaseAnalytics;

import net.maiatoday.levelbest.helpers.PreferenceHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Application Module for dependency injection
 * Created by maia on 2016/07/25.
 */
@Module
public class ApplicationModule {

    private final Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Context providesContext() {
        return application;
    }

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences(Context context) {
        return context.getSharedPreferences(PreferenceHelper.SHARED_PREF_NAME, Context.MODE_PRIVATE);
    }

    @Provides
    @Singleton
    FirebaseAnalytics provideAnalytics(Context context) {
        return FirebaseAnalytics.getInstance(context);
    }
}

File: mobile/src/main/kotlin/net/maiatoday/levelbest/di/ApplicationModule.kt
package net.maiatoday.levelbest.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

import com.google.firebase.analytics.FirebaseAnalytics

import net.maiatoday.levelbest.helpers.PreferenceHelper

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Application Module for dependency injection
 * Created by maia on 2016/07/25.
 */
@Module
class ApplicationModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun providesContext(): Context {
        return application
    }

    @Provides
    @Singleton
    internal fun provideSharedPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences(PreferenceHelper.SHARED_PREF_NAME, Context.MODE_PRIVATE)
    }

    @Provides
    @Singleton
    internal fun provideAnalytics(context: Context): FirebaseAnalytics {
        return FirebaseAnalytics.getInstance(context)
    }
}